### PR TITLE
Resolves IDENT-3414 ConnectionManager  refactor

### PR DESCRIPTION
**Pull request by acuderman**

### CODE REVIEW

File: aries_cloudagent/protocols/connections/v1_0/manager.py [hunk 1]
```diff
@@ -840,13 +868,13 @@ async def get_endpoints(self, conn_id: str) -> Tuple[str, str]:
...
 
         """
-        connection = await ConnRecord.retrieve_by_id(self._session, conn_id)
-
-        wallet = self._session.inject(BaseWallet)
-        my_did_info = await wallet.get_local_did(connection.my_did)
+        async with self.profile.session() as session:
+            connection = await ConnRecord.retrieve_by_id(session, conn_id)
+            wallet = session.inject(BaseWallet)
+            my_did_info = await wallet.get_local_did(connection.my_did)
```
Comment (vidp1) [inline aries_cloudagent/protocols/connections/v1_0/manager.py:874]: doesn0t this need to be declared 1 scope above?

Comment (acuderman) [reply aries_cloudagent/protocols/connections/v1_0/manager.py:874]: Do you mean my_did_info?

Python allows this

```
if True:
  y = 10

print(y) -> 10
```

So basically function is the scope in which all vars are declared

Comment (vidp1) [reply aries_cloudagent/protocols/connections/v1_0/manager.py:874]: Evil! :) Resolving.

**Review by vidp1 — APPROVED** [review]

**Comment by MaticDiba** [comment]
In `routes.py:connections_accept_request` we are still using sessions. Is that on purpose?

**Comment by acuderman** [comment]
> connections_accept_request

No missed this one. Thanks will fix It

**Review by MaticDiba — APPROVED** [review]

**Pull request merged**